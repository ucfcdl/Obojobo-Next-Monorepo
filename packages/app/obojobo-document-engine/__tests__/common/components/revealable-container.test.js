import React from 'react'
import RevealableContainer from '../../../src/scripts/common/components/revealable-container'
import TestRenderer from 'react-test-renderer'

describe('RevealableContainer', () => {
	test('RevealableContainer renders correctly', () => {
		const testRenderer = TestRenderer.create(
			<RevealableContainer
				className="class-name"
				onClick={jest.fn()}
				onDeleteButtonClick={jest.fn()}
				onDeleteButtonKeyDown={jest.fn()}
				maxWidth={100}
				isSelected={true}
			/>
		)
		const tree = testRenderer.toJSON()
		expect(tree).toMatchSnapshot()
	})

	test('RevealableContainer renders correctly without className', () => {
		const testRenderer = TestRenderer.create(
			<RevealableContainer
				onClick={jest.fn()}
				onDeleteButtonClick={jest.fn()}
				onDeleteButtonKeyDown={jest.fn()}
				maxWidth={100}
				isSelected={false}
			/>
		)
		const tree = testRenderer.toJSON()
		expect(tree).toMatchSnapshot()
	})

	test('All methods are called as expected', () => {
		const onDeleteButtonClick = jest.fn()
		const onDeleteButtonKeyDown = jest.fn()
		const onClick = jest.fn()

		const testRenderer = TestRenderer.create(
			<RevealableContainer
				className="class-name"
				onClick={onClick}
				onDeleteButtonClick={onDeleteButtonClick}
				onDeleteButtonKeyDown={onDeleteButtonKeyDown}
				maxWidth={100}
				isSelected={true}
			/>
		)

		expect(onDeleteButtonClick).not.toBeCalled()
		expect(onDeleteButtonKeyDown).not.toBeCalled()
		expect(onClick).not.toBeCalled()

		testRenderer.root
			.findByProps({ className: 'obojobo-draft--components--revealable-container class-name' })
			.props.onClick()

		expect(onDeleteButtonClick).not.toBeCalled()
		expect(onDeleteButtonKeyDown).not.toBeCalled()
		expect(onClick).toBeCalledTimes(1)

		testRenderer.root.findByType('button').props.onClick()

		expect(onDeleteButtonClick).toBeCalledTimes(1)
		expect(onDeleteButtonKeyDown).not.toBeCalled()
		expect(onClick).toBeCalledTimes(1)

		testRenderer.root.findByType('button').props.onKeyDown()

		expect(onDeleteButtonClick).toBeCalledTimes(1)
		expect(onDeleteButtonKeyDown).toBeCalledTimes(1)
		expect(onClick).toBeCalledTimes(1)
	})
})
