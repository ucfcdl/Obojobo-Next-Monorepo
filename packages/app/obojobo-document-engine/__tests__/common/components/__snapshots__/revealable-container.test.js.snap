// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RevealableContainer RevealableContainer renders correctly 1`] = `
<div
  className="obojobo-draft--components--revealable-container-wrapper  is-selected"
  contentEditable={false}
  style={
    Object {
      "maxWidth": "100px",
    }
  }
>
  <div
    className="obojobo-draft--components--button is-not-dangerous align-center delete-button"
    contentEditable={false}
  >
    <button
      className="button"
      contentEditable={false}
      onClick={[MockFunction]}
      onKeyDown={[MockFunction]}
      tabIndex={0}
    >
      ×
    </button>
  </div>
  <div
    className="obojobo-draft--components--revealable-container class-name"
    onClick={[MockFunction]}
  />
</div>
`;

exports[`RevealableContainer RevealableContainer renders correctly without className 1`] = `
<div
  className="obojobo-draft--components--revealable-container-wrapper  is-not-selected"
  contentEditable={false}
  style={
    Object {
      "maxWidth": "100px",
    }
  }
>
  <div
    className="obojobo-draft--components--button is-not-dangerous align-center delete-button"
    contentEditable={false}
  >
    <button
      className="button"
      contentEditable={false}
      onClick={[MockFunction]}
      onKeyDown={[MockFunction]}
      tabIndex={-1}
    >
      ×
    </button>
  </div>
  <div
    className="obojobo-draft--components--revealable-container"
    onClick={[MockFunction]}
  />
</div>
`;
